pkg/collector/providers/etcd.go

```diff
@@ -20,10 +21,16 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/config"
 )
 
+type EtcdBackend interface {
```

Comment:
Why do we need a new type here? Also all public types need to be commented, but I don't think this needs to be public, so if we do need it you can just `E` --> `e`

Reply:
Correct, I changed to `e`. We need it per [Consul](https://github.com/DataDog/datadog-agent/blob/0028f3eb92956bb9e979f32ee7bdf2abfefe3362/pkg/collector/providers/consul.go#L25) and [ZK](https://github.com/DataDog/datadog-agent/blob/0028f3eb92956bb9e979f32ee7bdf2abfefe3362/pkg/collector/providers/zookeeper.go#L26) as it makes tests much more easy to implement. Just use what you need.